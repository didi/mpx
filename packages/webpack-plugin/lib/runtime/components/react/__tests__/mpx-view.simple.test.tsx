import React from 'react'
import renderer from 'react-test-renderer'

// Mock dependencies to avoid complex type issues
jest.mock('@mpxjs/utils', () => ({
  error: jest.fn(),
  isFunction: jest.fn((fn) => typeof fn === 'function')
}))

jest.mock('../utils', () => ({
  parseUrl: jest.fn(),
  PERCENT_REGEX: /%$/,
  splitStyle: jest.fn((style) => ({ textStyle: {}, backgroundStyle: {}, innerStyle: style || {} })),
  splitProps: jest.fn((props) => ({ textProps: {}, innerProps: props })),
  useTransformStyle: jest.fn((style) => ({
    normalStyle: style || {},
    hasSelfPercent: false,
    hasPositionFixed: false,
    hasVarDec: false,
    varContextRef: { current: {} },
    setWidth: jest.fn(),
    setHeight: jest.fn()
  })),
  useHover: jest.fn(() => ({
    isHover: false,
    gesture: {}
  })),
  wrapChildren: jest.fn((props) => props.children),
  useLayout: jest.fn(() => ({
    layoutRef: { current: null },
    layoutStyle: {},
    layoutProps: {}
  })),
  extendObject: jest.fn((...args) => Object.assign({}, ...args))
}))

jest.mock('../getInnerListeners', () => ({
  __esModule: true,
  default: jest.fn((props) => props),
  getCustomEvent: jest.fn((type, evt, ref, props) => ({ type, target: {} }))
}))

jest.mock('../useNodesRef', () => ({
  __esModule: true,
  default: jest.fn()
}))

// Import after mocks
import View from '../mpx-view'

describe('View Component (Simple react-test-renderer)', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('should render without crashing', () => {
    const component = renderer.create(
      <View>Simple Test</View>
    )

    expect(component).toBeTruthy()
    const tree = component.toJSON()
    expect(tree).toBeTruthy()
  })

  it('should render children correctly', () => {
    const component = renderer.create(
      <View>
        <View>Child 1</View>
        <View>Child 2</View>
      </View>
    )

    const tree = component.toJSON()
    expect(tree).toBeTruthy()

    // Basic check that component rendered
    expect(tree).not.toBeNull()
  })

  it('should handle style prop', () => {
    const testStyle = {
      backgroundColor: 'red',
      padding: 10
    }

    const component = renderer.create(
      <View style={testStyle}>Styled View</View>
    )

    const tree = component.toJSON()
    expect(tree).toBeTruthy()
  })

  it('should handle empty props', () => {
    const component = renderer.create(<View />)

    const tree = component.toJSON()
    expect(tree).toBeTruthy()
  })

  it('should work with test renderer instance methods', () => {
    const component = renderer.create(
      <View>Test Instance</View>
    )

    const instance = component.root
    expect(instance).toBeTruthy()

    // Test that we can find the component
    try {
      const foundComponents = instance.findAllByType('div')
      // Since our mock returns 'div', this should work
      expect(Array.isArray(foundComponents)).toBe(true)
    } catch (error) {
      // If it fails, that's also okay for this simple test
      console.log('Expected behavior with mock components')
    }
  })

  it('should handle complex nested structure', () => {
    const component = renderer.create(
      <View style={{ flex: 1 }}>
        <View style={{ height: 50 }}>
          Header Content
        </View>
        <View style={{ flex: 1 }}>
          <View>Nested Content 1</View>
          <View>Nested Content 2</View>
        </View>
        <View style={{ height: 50 }}>
          Footer Content
        </View>
      </View>
    )

    const tree = component.toJSON()
    expect(tree).toBeTruthy()

    // Create a snapshot for this test
    expect(tree).toMatchSnapshot()
  })

  it('should handle updates correctly', () => {
    const component = renderer.create(
      <View>Initial Content</View>
    )

    let tree = component.toJSON()
    expect(tree).toBeTruthy()

    // Update the component
    component.update(
      <View>Updated Content</View>
    )

    tree = component.toJSON()
    expect(tree).toBeTruthy()
  })
})
