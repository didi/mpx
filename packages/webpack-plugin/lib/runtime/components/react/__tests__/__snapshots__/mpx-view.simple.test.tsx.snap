// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Component (Simple react-test-renderer) should handle complex nested structure 1`] = `
<div
  style={
    {
      "flex": 1,
    }
  }
>
  <div
    style={
      {
        "height": 50,
      }
    }
  >
    Header Content
  </div>
  <div
    style={
      {
        "flex": 1,
      }
    }
  >
    <div
      style={{}}
    >
      Nested Content 1
    </div>
    <div
      style={{}}
    >
      Nested Content 2
    </div>
  </div>
  <div
    style={
      {
        "height": 50,
      }
    }
  >
    Footer Content
  </div>
</div>
`;
